projects/ag-grid/documentation/code-examples/data-entry-grid/data-manager-added/data-manager-data-entry-grid-docs-demo-view-grid.component.ts

```diff
@@ -0,0 +1,330 @@
+
+  private _items: any[] = [];
+
+  constructor(
+    private agGridService: SkyAgGridService,
+    private changeDetector: ChangeDetectorRef,
+    private dataManagerService: SkyDataManagerService
+  ) {
+    this.noRowsTemplate = `<div class="sky-deemphasized">No results found.</div>`;
+  }
+
+  public ngOnInit(): void {
+    this.displayedItems = this._items;
+
+    this.dataManagerService.initDataView(this.viewConfig);
+
+    this.gridOptions = this.agGridService.getGridOptions({
+      gridOptions: {
+        columnDefs: this.columnDefs,
+        onGridReady: this.onGridReady.bind(this),
+      },
+    });
+
+    this.dataManagerService
+      .getDataStateUpdates(this.viewId)
+      .subscribe((state) => {
+        this.dataState = state;
+        this.setInitialColumnOrder();
+        this.updateData();
+        this.changeDetector.detectChanges();
+      });
+
+    this.dataManagerService.getActiveViewIdUpdates().subscribe((id) => {
+      this.isActive = id === this.viewId;
+      this.changeDetector.detectChanges();
+    });
+  }
+
+  public updateData(): void {
+    this.sortItems();
+    this.displayedItems = this.filterItems(this.searchItems(this._items));
+
+    if (this.dataState.onlyShowSelected) {
+      this.displayedItems = this.displayedItems.filter((item) => item.selected);
+    }
+
+    if (this.displayedItems.length > 0) {
+      this.gridApi?.hideOverlay();
+    } else {
+      this.gridApi?.showNoRowsOverlay();
+    }
+
+    this.changeDetector.markForCheck();
+  }
+
+  public setInitialColumnOrder(): void {
+    let viewState = this.dataState.getViewStateById(this.viewId);
+    let visibleColumns = viewState.displayedColumnIds;
+
+    this.columnDefs.sort((col1, col2) => {
+      let col1Index = visibleColumns.findIndex(
+        (colId: string) => colId === col1.colId
+      );
+      let col2Index = visibleColumns.findIndex(
+        (colId: string) => colId === col2.colId
+      );
+
+      if (col1Index === -1) {
+        col1.hide = true;
+        return 0;
+      } else if (col2Index === -1) {
+        col2.hide = true;
+        return 0;
+      } else {
+        return col1Index - col2Index;
+      }
+    });
+
+    this.gridInitialized = true;
+    this.changeDetector.markForCheck();
+  }
+
+  public onGridReady(gridReadyEvent: GridReadyEvent): void {
+    this.gridApi = gridReadyEvent.api;
+    this.gridApi.sizeColumnsToFit();
+    this.columnApi = gridReadyEvent.columnApi;
+    this.updateData();
+    this.changeDetector.markForCheck();
+  }
+
+  public onRowSelected(rowSelectedEvent: RowSelectedEvent): void {
+    if (!rowSelectedEvent.data.selected) {
+      this.updateData();
+    }
+  }
+
+  public sortItems(): void {
+    let sortOption = this.dataState.activeSortOption;
+    if (this.columnApi && sortOption) {
+      const sort = [
+        {
+          colId: sortOption.propertyName,
+          sort: sortOption.descending ? 'desc' : 'asc',
+        },
+      ];
+      this.gridApi?.setSortModel(sort);
```

Comment:
Again, I'm seeing errors
```
AG Grid: as of version 24.0.0, setSortModel() is deprecated, sort information is now part of Column State. Please use columnApi.applyColumnState() instead.
```

Reply:
fixed in commit [17030aa - _"fix sort deprecation warning"_](https://github.com/blackbaud/skyux-ag-grid/pull/194/commits/17030aaa1a2498cd47cc247bbed2d42bb5ad7f88)